src/reference-data/utils.ts

```diff
@@ -30,3 +35,10 @@ export const getReferenceDataName = <
   {name, alternativeNames}: T,
   language: Language,
 ) => getReferenceDataText([name, ...(alternativeNames || [])], language);
+
+export const findReferenceDataById = <
+  T extends UserProfile | PreassignedFareProduct | TariffZone
+>(
+  elements: T[],
+  id: string,
```

Comment:
Technically this could be undefined. Although it won't change the implementation as is

Reply:
You mean the return value? Or the `id` parameter?

Reply:
ID parameter (I think). At some point, the passed argument is `someArray[0]` and `someOtherArray.slice(-1)`. Unless we actually know the arrays have the "correct" amount of items.

Reply:
My thought was that the tariff zones array always was non-empty, as it does not make sense to have a ticket with no tariff zones, and if so both `someArray[0]` and `someOtherArray.slice(-1)` will never be undefined. But as seen in the ongoing chat on Slack these assumptions should be confirmed. And the type should probably then be `[string, ...string[]]` instead of `string[]`.